app/controllers/tag_controller.rb

```diff
@@ -359,7 +359,7 @@ def delete
     node_tag = NodeTag.where(nid: params[:nid], tid: params[:tid]).first
     node = Node.where(nid: params[:nid]).first
     # only admins, mods, and tag authors can delete other peoples' tags
-    if node_tag.uid == current_user.uid || logged_in_as(['admin', 'moderator']) || node.uid == current_user.uid
+    if node_tag.uid == current_user.uid || logged_in_as(['admin', 'moderator']) || (node.uid == current_user.uid && node_tag.name != "locked")
```

Comment:
From line 359 above, we have the `node_tag` variable derived from this query `node_tag = NodeTag.where(nid: params[:nid], tid: params[:tid]).first` ,

 I am curious as to whether checking to see if `node_tag.name != "locked"` will work if the first tag derived is not `"locked"` but contains a list of tags like `[air-quality, locked]`....the node itself does contain a locked tag but it may not be the first to be picked hence someone else might delete a tag.

Should checking if the tags returned contain `"locked"`, before picking the first one in line 359 help?

Reply:
lol I was just reviewing this and panicked that I had merged it @jywarren 😅 

Are the concerns I have above okay? Or will the tagging work as expected... @jywarren?

Reply:
Oh you're totally right, my bad! Yes, let's change this, we can use `@node.has_tag('locked')` instead!

Reply:
@Manasa2850 what do you say? Thanks all!